src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c

```diff
@@ -843,6 +843,10 @@ create_stream_with_ctx (
 mongoc_stream_t *
 mongoc_stream_tls_openssl_new (mongoc_stream_t *base_stream, const char *host, mongoc_ssl_opt_t *opt, int client)
 {
+   if (!opt) {
+      RETURN (NULL);
+   }
```

Comment:
Should this check `opt` or assert `opt`?

Reply:
Same as the other instance of checking `opt`. The assert seems to work (and is preferrable), so commit incoming.